packages/sdk/src/evm/core/classes/staking-20.ts

```diff
@@ -0,0 +1,448 @@
+   * @twfeature Staking20
+   */
+  public async getTokensStaked(address: AddressOrEns): Promise<CurrencyValue> {
+    return fetchCurrencyValue(
+      await this.contractWrapper.getProvider(),
+      (await this.getStakingToken()).address,
+      (
+        await this.contractWrapper.readContract.getStakeInfo(
+          await resolveAddress(address),
+        )
+      )[0],
+    );
+  }
+
+  /**
+   * Get the unclaimed rewards for a particular wallet address or ENS.
+   *
+   * @example
+   * ```javascript
+   * // Get unclaimed rewards
+   * const unclaimedRewards = await contract.staking20.getRewards("thirdweb.eth");
+   * ```
+   *
+   * @returns Number of unclaimed reward tokens, see {@link CurrencyValue}
+   * @twfeature Staking20
+   */
+  public async getRewards(address: AddressOrEns): Promise<CurrencyValue> {
+    return fetchCurrencyValue(
+      await this.contractWrapper.getProvider(),
+      (await this.getRewardToken()).address,
+      (
+        await this.contractWrapper.readContract.getStakeInfo(
+          await resolveAddress(address),
+        )
+      )[1],
+    );
+  }
+
+  // TODO: Create a type
+
+  /**
+   * Get the time unit for new reward tokens set in the staking contract.
+   *
+   * @example
+   * ```javascript
+   * // Get time unit for staking contract
+   * const rewardRatio = await contract.staking20.getRewardRatio();
+   * ```
+   *
+   * @returns Reward ratio numerator and denominator
+   * ```javascript
+   * {
+   *    numerator: 1,
+   *    denominator: 2
+   * }
+   * ```
+   *
+   * @twfeature Staking20
+   */
+  public async getRewardRatio(): Promise<{
+    numerator: number;
+    denominator: number;
+  }> {
+    return {
+      numerator: (
+        await this.contractWrapper.readContract.getRewardRatio()
+      )[0].toNumber(),
+      denominator: (
+        await this.contractWrapper.readContract.getRewardRatio()
+      )[1].toNumber(),
+    };
+  }
+
+  // TODO: Create types
+
+  /**
+   * Get information about staking token
+   *
+   * @example
+   * ```javascript
+   * // Get staking token info
+   * const stakingToken = await contract.staking20.getStakingToken();
+   * ```
+   *
+   * @returns see {@link Currency}
+   * @twfeature Staking20
+   */
+  public async getStakingToken(): Promise<Currency & { address: string }> {
+    return {
+      ...(await fetchCurrencyMetadata(
+        await this.contractWrapper.getProvider(),
+        await this.contractWrapper.readContract.stakingToken(),
+      )),
+      address: await this.contractWrapper.readContract.stakingToken(),
+    };
+  }
+
+  // TODO: Create types
+
+  /**
+   * Get information about reward token
+   *
+   * @example
+   * ```javascript
+   * // Get reward token info
+   * const rewardToken = await contract.staking20.getRewardToken();
+   * ```
+   *
+   * @returns see {@link Currency}
+   * @twfeature Staking20
+   */
+  public async getRewardToken(): Promise<Currency & { address: string }> {
```

Comment:
can you create an explicit type for this return type? `CurrencyWithAddress` or something